scipy/stats/_covariance.py

```diff
@@ -172,6 +172,72 @@ def _covariance(self):
                 if self._cov_matrix is None else self._cov_matrix)
 
 
+def _dot_diag(x, d):
+    # If d were a full diagonal matrix, x @ d would always do what we want
+    # This is for when `d` is compressed to include only the diagonal elements
+    return x * d if x.ndim < 2 else x * np.expand_dims(d, -2)
```

Comment:
Do we need to have the `else` statement here i.e. why do we need to add an extra dimension to `d` when `x` is multi-dimensional? Wouldn't NumPy right broadcast and add the necessary dimensions for us?

Reply:
This is needed when `d` is n-dimensional. I didn't want to add those tests yet. Sure, for now I'll remove it.

Reply:
Oh, ok. That makes sense. Thanks for clarifying! I didn't get that from the comments. We can keep it and just add a comment that it will handle the multidimensional parameter case when we add support for it.

Reply:
I think removing it now and adding it back later is easier than adding the comment now and remembering to remove the comment later, tbh.

Reply:
NM I understand - I'll just clarify the coment.

Reply:
Thanks!